This test verifies that nested #anySatisfy: blocks do not throw an exception.